ext/dom/nodelist.c

```diff
@@ -154,9 +154,9 @@ PHP_METHOD(DOMNodeList, item)
 	int count = 0;
 
 	id = ZEND_THIS;
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "l", &index) == FAILURE) {
-		RETURN_THROWS();
-	}
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_LONG(index)
+	ZEND_PARSE_PARAMETERS_END();
```

Comment:
Unrelated, is it me or does this just do nothing when passing a negative index? :| Does the spec allow for asking a negative index or should this be changed to emit a warning?

Reply:
Yeah I also noticed this, and I'm afraid we can't change the behaviour without making a BC break :(
Spec link for item: https://www.w3.org/TR/2003/WD-DOM-Level-3-Core-20030226/DOM3-Core.html#core-ID-349467F9
The problem is that spec defines the index as an unsigned type...
`If index is greater than or equal to the number of nodes in this map, this returns null. `
and further down: `The node at the indexth position in the map, or null if that is not a valid index.`
So it sounds to me it shouldn't throw, but rather should return NULL.

Reply:
Right, normally what we do is that we used to emit an E_WARNING for negative indices, which got changed to ValueErrors in PHP 8.0.

I think changing this to emit a warning for negative indices is at least a good start to indicate there is some mistake. But that's another PR to be done, (and probably check that we are handling indices consistently too...).

Reply:
OTOH, people coming from only knowing the DOM spec will be surprised that this throws. Spec doesn't define exceptions for this.
I tried with javascript, try executing this: `document.getElementsByTagName('p').item(-1)` it just returns NULL without an exception.

Reply:
Huh, welp IDK then. But in the other method we do check for the offset so :|